sale_report_delivered_subtotal/reports/sale_report.py

```diff
@@ -14,10 +14,12 @@ def _query(self, with_clause="", fields=None, groupby="", from_clause=""):
         if fields is None:
             fields = {}
         select_str_price = """ ,
-            sum((l.price_subtotal /
-                 coalesce(nullif(l.product_uom_qty, 0), 1)
-                ) * l.qty_delivered)
-            as price_subtotal_delivered
+        sum(
+            CASE WHEN l.price_subtotal <> 0
+                THEN (l.price_subtotal / l.product_uom_qty)
+                ELSE l.price_reduce
```

Comment:
Why not doing here the operation of price_unit - discount for getting a price reduced without rounding?

Reply:
Because for these rare cases I prefer the result to be more compatible with other modules than more accurate.